src/main/java/codec/Codec.java

```diff
@@ -37,21 +37,21 @@ public ByteBuffer encode(Message message) throws Exception {
 
             return messageBuffer;
         } else {
-            throw new Exception("Max message size = " + MAX_LENGTH + " | Your message size is " + message.getMessage().length());
+            throw new IndexOutOfBoundsException("Max message size = " + MAX_LENGTH + " | Your message size is " + message.getMessage().length());
         }
     }
 
     public boolean canDecode(ByteBuffer buffer) {
-        try {
-            buffer.rewind();
-            int fullSizeMessage = buffer.getInt();
-            byte[] tempBytes = new byte[fullSizeMessage];
-            buffer.rewind();
-            buffer.get(tempBytes);
-            return true;
-        } catch (Exception e) {
+        if (buffer.remaining() < 4) {
             return false;
         }
+        int fullSizeMessage = buffer.getInt();
+        buffer.rewind();
+        if (fullSizeMessage < 1 || fullSizeMessage > buffer.limit()) {
+            return false;
+        }
+        buffer.position(fullSizeMessage);
```

Comment:
Why do you set the value read from the buffer as a position? Is there any guarantee that the `buffer` you are reading has the start position at 0?